### PR TITLE
Add gateway service for poet

### LINKED ISSUE #70: Add support for poet broadcaster
The poet server currently relies upon the [`BroadcastPoet` endpoint](https://github.com/spacemeshos/poet/blob/4c0687228ed4ff7ab2353800da6e4d9ede60fce7/broadcaster/pb/api.proto#L76-L81) in the old API. Figure out how to support this in the new API.

Existing implementation: https://github.com/spacemeshos/go-spacemesh/blob/a497f1817a0a368e801a9c7f44c6b16bc848dad3/api/grpc_server.go#L214-L223

### PR DESCRIPTION
**Pull request by lrettig**
Closes #70 

This provides a basic endpoint that allows the poet to broadcast a message using the new API. It's basically the same as the old API. I put it into a new service since it doesn't really belong in any of the other services, and since we may want to add other gateway-related services.

### CODE REVIEW

**Comment by avive** [comment]
I think that @moshababo should review as he's closer to this than me.

**Comment by lrettig** [comment]
CI failure is expected here as the interface is changing, will override when merging

File: README.md [hunk 1]
```diff
@@ -40,11 +40,12 @@ Broadly speaking, there are four types of endpoints: simple, command, query, and
 
 The Spacemesh API consists of several logical services, each of which contains a set of one or more RPC endpoints. The node operator can enable or disable each service independently using the CLI. The current set of services is as follows:
 
-- [NodeService](/proto/spacemesh/v1/node.proto) is a readonly interface for reading basic node-related data such as node status, software version and build number, and errors. It also allows a consumer to request that the node start the sync process, thus enabling the stream endpoints.
-- [MeshService](/proto/spacemesh/v1/mesh.proto) is a readonly interface that provides access to mesh data such as layer number, epoch number, and network ID. It provides streams for watching layers (which contain blocks, transactions, etc.). In the future this service will be expanded to include other mesh-related endpoints.
+- [GatewayService](/proto/spacemesh/v1/gateway.proto) is a read-write interface that allows a [poet server](https://github.com/spacemeshos/poet) to broadcast data to the network via a poet gateway node.
```
Comment (moshababo) [inline README.md:43]: Perhaps: "... allows a poet server to broadcast proofs to the network."

Comment (moshababo) [inline README.md:43]: Perhaps: "... allows a poet server to broadcast proofs to the network."

File: proto/spacemesh/v1/gateway_types.proto [hunk 2]
```diff
@@ -0,0 +1,13 @@
+syntax = "proto3";
+package spacemesh.v1;
+option go_package = "github.com/spacemeshos/api/release/go/spacemesh/v1";
+import "google/rpc/status.proto";
+
+message BroadcastPoetRequest {
+    bytes data = 1;  // signed binary transaction
```
Comment (moshababo) [inline proto/spacemesh/v1/gateway_types.proto:7]: This one isn't a transaction, but an XDR-encoded data of `PoetProofMessage` defined in `github.com/spacemeshos/poet/service` Go package.

**Pull request merged**